jwk.go

```diff
@@ -216,7 +216,7 @@ func (k *JSONWebKey) UnmarshalJSON(data []byte) (err error) {
 		}
 		key, err = raw.symmetricKey()
 	case "OKP":
-		if raw.Crv == "Ed25519" && raw.X != nil {
+		if raw.Crv == "Ed25519" {
```

Comment:
Why does this PR remove the `X != nil` check?

Reply:
Had to ask that question myself too. I don't have an exact recollection of the reason I had when writing this, but I think I simply felt the check is in the wrong place. The function calls to `raw.edPrivateKey()` and `raw.edPublicKey()` return an error if X is missing, and I feel those functions are the correct places to validate the key internals.